components/math/multiplicationConnect/GameBoardBlock.tsx

```diff
@@ -1,20 +1,20 @@
 import React, {useState} from 'react'
+import { calculateWinner } from '../../../pages/studentPortal/labs/multiplication-connect/Index';
 
 
-const blockClick = (block) => {
-    (block.isSelected===false) ? block.isSelected=true : block.isSelected=false;
-    // console.log(block);
+const blockClick = (block, grid) => {
+    (block.isSelected===false) ? (block.isSelected=true, calculateWinner(block, grid)) : block.isSelected=false;
```

Comment:
It's hard to read what this function is trying to do

Reply:
On a block click, it's isSelected state is being toggled. If it's toggled from false to true I have it run the calculateWinner algorithm to check for a game win. Does that make sense or should I switch things up?